examples/cosmos/ics20/ics20.qnt

```diff
@@ -0,0 +1,129 @@
+// -*- mode: Bluespec; -*-
+
+/**
+ * A specification ICS20 fungible token transfer protocol.
+ *
+ * For now, this only models the denomination trace changes between transfers.
+ *
+ * Gabriela Moreira, Informal Systems, 2023
+ */
+module ics20 {
+  type Uint = int
+
+  type State = {
+    trace: List[str],
+    denom: str,
+  }
+
+  type FungibleTokenPacketData = {
+    trace: List[str],
+    denom: str,
```

Comment:
These two fields are packed into `denom` in [ICS20](https://github.com/cosmos/ibc/blob/main/spec/app/ics-020-fungible-token-transfer/README.md#data-structures). Shall we simply use `State` aka `IbcDenom` from the above type definition?

Reply:
Also, `str` inside `List` are actually channel or port identifiers, so it would help to have a record there, e.g.. `{ channel: str, port: str }`. Our first goal is not to mirror the ICS/go/rust data structures one-to-one, but to keep the concepts simple.

Reply:
sounds much better, will do!